src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -66,6 +74,74 @@ public override MongoDatabaseSettings Settings
         }
 
         // public methods
+        public override IAsyncCursor<TResult> Aggregate<TResult>(PipelineDefinition<DatabaseAggregateVoid, TResult> pipeline, AggregateOptions options, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
These are all similar to the equivalent collection methods.

I looked into reducing code duplication, but it's not really possible. Every few lines there is a subtle difference that precludes easy sharing.

Reply:
Yes, agree. I didn't notice that a number of code lines have different types, even though they look almost the same in the editor